### PR TITLE
New Table

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

# Preview

# https://deploy-preview-19678--ant-design.netlify.com/components/table-cn/#header

-----
[View rendered components/table/demo/ajax.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/ajax.md)
[View rendered components/table/demo/bordered.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/bordered.md)
[View rendered components/table/demo/colspan-rowspan.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/colspan-rowspan.md)
[View rendered components/table/demo/custom-filter-panel.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/custom-filter-panel.md)
[View rendered components/table/demo/drag-sorting.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/drag-sorting.md)
[View rendered components/table/demo/dynamic-settings.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/dynamic-settings.md)
[View rendered components/table/demo/edit-cell.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/edit-cell.md)
[View rendered components/table/demo/edit-row.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/edit-row.md)
[View rendered components/table/demo/ellipsis.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/ellipsis.md)
[View rendered components/table/demo/expand-children.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/expand-children.md)
[View rendered components/table/demo/expand.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/expand.md)
[View rendered components/table/demo/fixed-columns-header.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/fixed-columns-header.md)
[View rendered components/table/demo/fixed-columns.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/fixed-columns.md)
[View rendered components/table/demo/fixed-header.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/fixed-header.md)
[View rendered components/table/demo/grouping-columns.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/grouping-columns.md)
[View rendered components/table/demo/multiple-sorter.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/multiple-sorter.md)
[View rendered components/table/demo/nested-table.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/nested-table.md)
[View rendered components/table/demo/reset-filter.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/reset-filter.md)
[View rendered components/table/demo/resizable-column.md](https://github.com/ant-design/ant-design/blob/4.0-table/components/table/demo/resizable-column.md)

### CODE REVIEW

**Comment by afc163** [comment]
image

在 Firefox 下没对齐。

**Comment by afc163** [comment]
image

横向滚动在 Firefox 下看明确比 3.x 要卡顿。

File: components/table/style/index.less [hunk 1]
```diff
@@ -344,6 +344,7 @@
     .operation-unit();
     position: relative;
     display: inline-block;
+    display: inline-flex;
```
Comment (afc163) [inline components/table/style/index.less]: inline-block; 是不是应该删了。

**Comment by afc163** [comment]
image

hover 到菜单时，图标背景高亮不会丢失。

**Comment by afc163** [comment]
image

Blur 这个框时，消失的时候浮层会闪动，3.x 没有这个问题。

**Comment by zombieJ** [comment]
> image
> 
> Blur 这个框时，消失的时候浮层会闪动，3.x 没有这个问题。

@shaodahong, Please help to check on this. Seems dropdown blink with new update on Firefox. Mini reproduce:

```jsx
const inputRef = React.createRef();

const menu = (
  <div style={{ padding: 16, background: '#FFF', border: '1px solid red' }}>
    <Input ref={inputRef} />
  </div>
);

ReactDOM.render(
  <Dropdown
    overlay={menu}
    trigger={['click']}
    onVisibleChange={visible => {
      if (visible) {
        setTimeout(() => inputRef.current.select());
      }
    }}
  >
    <a className="ant-dropdown-link" href="#">
      Click me <Down />
    </a>
  </Dropdown>,
  mountNode,
);
```

**Comment by zombieJ** [comment]
# Virtual Scroll
https://deploy-preview-19678--ant-design.netlify.com/components/table-cn/#components-table-demo-virtual-list

**Comment by afc163** [comment]
image

- [ ] 发现这个 Demo 的编辑图标不见了。
- [ ] 这个文案太难看懂了，应该是加载 ****** 条数据的高性能表格。

**Pull request merged**